### PR TITLE
Replace Prometheus envsubst variables with text/template variables

### PR DESCRIPTION
**Pull request by jshufro**
So, the syntax here is two phrases.

First, `(index . "FOO")` when passed a `settings map[string]string` is the same as `settings["FOO"]`

Second, `or x y` is the same as `x ? x  : y`

Combining them gives us the defaulting behavior we had with envsubst.

In a future change, I suggest we move away from using `map[string]string` and instead just pass the config struct to the template engine. Then we can write variables such as `{{or .Execution.ApiPortNumber "8545"}}` which is a bit more readable.

### CODE REVIEW

**Comment by jclapis** [comment]
> In a future change, I suggest we move away from using `map[string]string` and instead just pass the config struct to the template engine. Then we can write variables such as `{{or .Execution.ApiPortNumber "8545"}}` which is a bit more readable.

Probably worth doing now honestly, readability does take a bit of a hit here. I'd set up https://github.com/rocket-pool/smartnode/pull/420 with the struct setup.

**Comment by jshufro** [comment]
> > In a future change, I suggest we move away from using `map[string]string` and instead just pass the config struct to the template engine. Then we can write variables such as `{{or .Execution.ApiPortNumber "8545"}}` which is a bit more readable.
> 
> Probably worth doing now honestly, readability does take a bit of a hit here. I'd set up [rocket-pool/smartnode#420](https://github.com/rocket-pool/smartnode/pull/420) with the struct setup.

I'll add commits on top that move that way and we can decide how much we want to bite off at once based on those

**Comment by jshufro** [comment]
@jclapis I updated this as well as https://github.com/rocket-pool/smartnode/pull/420

Note the extra complexity in smartnode. It looks like we're adding complexity but we're actually removing it- once every template is migrated, we can delete all the environment variable fudging, instead using these `cfg` receiver methods returning `(sprintable, error)`, so during the transition there is more complexity, but in the steady state there should be much less.

File: install/prometheus.tmpl [hunk 1]
```diff
@@ -8,42 +8,42 @@ global:
...
   - job_name: 'geth'
     static_configs:
-      - targets: ['${EC_HOSTNAME:-eth1}:${EC_METRICS_PORT:-9105}']
+      - targets: ['{{.GetExecutionHostname}}:{{or .EcMetricsPort.Value "9105"}}']
     metrics_path: /debug/metrics/prometheus
 
   - job_name: 'eth1'
     static_configs:
-      - targets: ['${EC_HOSTNAME:-eth1}:${EC_METRICS_PORT:-9105}']
+      - targets: ['{{.GetExecutionHostname}}:{{or .EcMetricsPort.Value "9105"}}']
```
Comment (jshufro) [inline install/prometheus.tmpl:26]: since we're using text/template now we could actually combine these into a single job, if you want

Comment (jshufro) [reply install/prometheus.tmpl]: (merged in HEAD to demonstrate)

File: install/templates/eth2.tmpl [hunk 2]
```diff
@@ -8,54 +8,57 @@
...
   eth2:
-    image: ${BN_CONTAINER_TAG}
+    image: {{.GetBeaconContainerTag}}
     user: root
-    container_name: ${COMPOSE_PROJECT_NAME}_eth2
+    container_name: {{.Smartnode.ProjectName.Value}}_eth2
     restart: unless-stopped
     stop_grace_period: 3m
-    ports: [ "${BN_P2P_PORT:-9001}:${BN_P2P_PORT:-9001}/tcp", "${BN_P2P_PORT:-9001}:${BN_P2P_PORT:-9001}/udp"${BN_OPEN_PORTS}, "${BN_P2P_QUIC_PORT:-8001}:${BN_P2P_QUIC_PORT:-8001}/udp"]
+    {{- $p2p := (or .ConsensusCommon.P2pPort.Value "9001")}}
```
Comment (0xfornax) [inline install/templates/eth2.tmpl:16]: It's not clear where the quic port is now.

Comment (jshufro) [reply install/templates/eth2.tmpl:16]: Looks like i missed it

**Pull request merged**